apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -446,7 +448,11 @@ public override bool NextResult()
                             }
                             // This excpetion may occur if Spanner had to split returned data into multiple PartialResultSets.
                             // This should not happen right now as DML statement can never execute more than 40,000 mutations but could increase in the future.
-                            catch (Exception ex) when (ex.Message.Contains("Read/query was cancelled due to Commit()/Rollback() on the enclosing transaction"))
+                            // TODO: Need to confirm behaviour of this catch block whenever we are able to produce the excpetion of multiple PartialResultSets.
+                            catch (Exception ex) when (
+                            ex is RpcException rpcException &&
+                            rpcException.GetErrorInfo() != null &&
+                            rpcException.GetErrorInfo().Reason.Contains("Read/query was cancelled due to Commit()/Rollback() on the enclosing transaction."))
```

Comment:
Hi @jskeet , does this seems fine ? If not, please let me know what do you have in you mind.

Reply:
I think based on latest Knut's comment, we may need to change this.

Reply:
Yes, the Reason would never be something like this - it would be an identifier, basically. I think it's worth consulting the Spanner term internally though - we should really try to avoid using text matching. It would be much better for the Spanner team to *start* using ErrorInfo if they're really not already.

Additionally, you shouldn't catch Exception and *then* check the type - just `catch (RpcException ex)` to start with.

Reply:
Sure. As we'll not get this particular exception right now, removing the catch block for now.